src/main/java/jenkins/plugins/slack/webhook/WebhookEndpoint.java

```diff
@@ -0,0 +1,245 @@
+package jenkins.plugins.slack.webhook;
+
+
+import jenkins.model.Jenkins;
+import jenkins.model.GlobalConfiguration;
+
+import hudson.Extension;
+
+import hudson.model.Build;
+import hudson.model.Result;
+import hudson.model.Project;
+import hudson.model.AbstractBuild;
+import hudson.model.AbstractProject;
+import hudson.model.UnprotectedRootAction;
+
+import hudson.security.ACL;
+
+import javax.servlet.ServletException;
+
+import java.io.IOException;
+
+import java.util.List;
+import java.util.ArrayList;
+
+import java.util.logging.Logger;
+
+import org.kohsuke.stapler.HttpResponse;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.StaplerResponse;
+
+import org.kohsuke.stapler.interceptor.RequirePOST;
+
+import jenkins.plugins.slack.webhook.model.JsonResponse;
+import jenkins.plugins.slack.webhook.model.SlackPostData;
+import jenkins.plugins.slack.webhook.model.SlackTextMessage;
+import jenkins.plugins.slack.webhook.model.SlackWebhookCause;
+
+import jenkins.plugins.slack.webhook.exception.CommandRouterException;
+import jenkins.plugins.slack.webhook.exception.RouteNotFoundException;
+
+
+
+
+@Extension
+public class WebhookEndpoint implements UnprotectedRootAction {
```

Comment:
Especially since it's an unprotected URL, I'm a little concerned that it's accessible just by installing the plugin. I know you need the token to match to be able to do anything, but there is still something listening on that URL even if the token is not set. It at least lets somebody determine that you have the Slack plugin installed on your system by trying to access /webhook. Any way to disable it unless the token is set?